src/dstruct.d

```diff
@@ -255,6 +256,7 @@ public:
         StructDeclaration sd =
             s ? cast(StructDeclaration)s
               : new StructDeclaration(loc, ident);
+        sd.cppmangleAsClass = cppmangleAsClass;
```

Comment:
This sticks in my craw, as no other attributes (like `extern (C++)`) behave that way. This is one of the motivating reasons to change it to `extern (C++, class)` instead of a pragma.